Sources/TuistDependenciesTesting/Carthage/Utils/MockCarthage.swift

```diff
@@ -0,0 +1,84 @@
+import TSCBasic
+import TuistGraph
+
+@testable import TuistDependencies
+
+public final class MockCarthage: Carthaging {
+    public init() {}
+
+    var invokedBootstrap = false
+    var invokedBootstrapCount = 0
+    var invokedBootstrapParameters: BootstrapParameters?
+    var invokedBootstrapParametersList = [BootstrapParameters]()
+    var bootstrapStub: ((BootstrapParameters) throws -> Void)?
+
+    public func bootstrap(at path: AbsolutePath, platforms: Set<Platform>?, options: Set<CarthageDependencies.Options>?) throws {
+        let parameters = BootstrapParameters(
+            path: path,
+            platforms: platforms,
+            options: options
+        )
+
+        invokedBootstrap = true
+        invokedBootstrapCount += 1
+        invokedBootstrapParameters = parameters
+        invokedBootstrapParametersList.append(parameters)
+        try bootstrapStub?(parameters)
+    }
+
+    var invokedUpdate = false
+    var invokedUpdateCount = 0
+    var invokedUpdateParameters: UpdateParameters?
+    var invokedUpdateParametersList = [UpdateParameters]()
+    var updateStub: ((UpdateParameters) throws -> Void)?
+
+    public func update(at path: AbsolutePath, platforms: Set<Platform>?, options: Set<CarthageDependencies.Options>?) throws {
+        let parameters = UpdateParameters(
+            path: path,
+            platforms: platforms,
+            options: options
+        )
+
+        invokedUpdate = true
+        invokedUpdateCount += 1
+        invokedUpdateParameters = parameters
+        invokedUpdateParametersList.append(parameters)
+        try updateStub?(parameters)
+    }
+}
+
+// MARK: - Models
+
+extension MockCarthage {
+    struct BootstrapParameters {
```

Comment:
this highlights my earlier comment pointing at the unwieldiness of the current stub method - it's also not even used. I'd try to keep the mocks simpler, honestly

Reply:
I'd rather not have these helper structs at all - but maybe it's just my opinion))

Reply:
it's also in multiple places below